libs/ardour/session.cc

```diff
@@ -7347,7 +7364,7 @@ Session::cancel_all_solo ()
 
 	get_stripables (sl);
 
-	set_controls (stripable_list_to_control_list (sl, &Stripable::solo_control), 0.0, Controllable::NoGroup);
+	// set_controls (stripable_list_to_control_list (sl, &Stripable::solo_control), 0.0, Controllable::NoGroup);
```

Comment:
There is probably still work to be done here. It looked like this line and the next one would do the same - but I'm not sure. When this line is executed, it seems that solo mode is already disabled in all tracks clear_all-solo_state then does nothing - so my mechanism would not work like that.